### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by kuldeepk3**
Signed-off-by: Kuldeep Kumar <kuldeep.kumar3@dell.com>

**Change log description**  
(2-3 concise points about the changes in this PR. When committing this PR, the committer is expected to copy the content of this section to the merge description box)

**Purpose of the change**  
(_e.g._, Fixes #666, Closes #1234)

**What the code does**  
(Detailed description of the code changes)

**How to verify it**  
(Optional: steps to verify that the changes are effective)

### CODE REVIEW

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java [hunk 1]
```diff
@@ -1573,6 +1576,85 @@ private <R> CompletableFuture<R> executeExclusive(Callable<CompletableFuture<R>>
         }, this.executor);
     }
 
+    public CompletableFuture<Void> saveEpochInfo(int containerId, long containerEpoch, ChunkedSegmentStorage chunkedSegmentStorage) {
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java:1579]: Don't need to pass this parameter.
SystemJournal has reference chunkStorage already

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java [hunk 2]
```diff
@@ -1573,6 +1576,85 @@ private <R> CompletableFuture<R> executeExclusive(Callable<CompletableFuture<R>>
...
+        } catch (IOException e) {
+            return CompletableFuture.failedFuture(new ChunkStorageException(chunk, "Unable to serialize", e));
+        }
+    }
+
+    private CompletableFuture<EpochInfo> readEpochInfo(String chunk, ChunkedSegmentStorage chunkedSegmentStorage, int readLength) {
+        val readAtOffset = new AtomicLong(0);
+        val readBuffer = new byte[readLength];
+        return chunkedSegmentStorage.getChunkStorage().read(ChunkHandle.readHandle(chunk), readAtOffset.get(), readLength, readBuffer, 0)
+                .thenApplyAsync( v -> {
```
Comment (sachin-j-joshi) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java:1648]: reference to chunkedSegmentStorage not needed.
use this.getChunkStorage() instead.

**Review by sachin-j-joshi — APPROVED** [review]
LGTM

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java (hunk 1 above)
Comment (kuldeepk3) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java]: updated

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java (hunk 2 above)
Comment (kuldeepk3) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java:1648]: done.

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java [hunk 3]
```diff
@@ -852,92 +846,13 @@ public CompletableFuture<Void> flushToStorage(Duration timeout) {
...
-        return chunkedSegmentStorage.getChunkStorage().read(ChunkHandle.readHandle(chunk), readAtOffset.get(), readLength, readBuffer, 0)
-                .thenApplyAsync( v -> {
-                    try {
-                        return EPOCH_INFO_SERIALIZER.deserialize(readBuffer);
-                    } catch (IOException e) {
-                        throw new CompletionException(e);
-                    }
-                }, this.executor);
-
+        return chunkedSegmentStorage.getSystemJournal().saveEpochInfo(containerId, this.metadata.getContainerEpoch())
```
Comment (sachin-j-joshi) [inline segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java:854]: Use flushed.thenComposeAsync . (remember to pass executer)

**Comment by tkaitchuck** [comment]
Please add an Issue and a description

**Comment by sachin-j-joshi** [comment]
@kuldeepk3 This needs to be retargeted to master.

**Review by sachin-j-joshi — CHANGES_REQUESTED** [review]
This needs to be retargeted to master.

**Comment by kuldeepk3** [comment]
Will reopen for master.

**Pull request closed without merging**